src/js/shared/config_file.js

```diff
@@ -212,11 +232,44 @@ const ConfigFile = {
     return [];
   },
 
-  parseSpecialProvider: function (storeKey, subData) {
-    return LocalStore.setOne(storeKey, {
-      config: _.get(subData, "Config", {}),
-      queries: _.map(_.get(subData, "Queries", []), ConfigFile.parseQuery),
-    });
+  parseSpecialProvider: async function (storeKey, newData, mergeKey) {
+    var settings = await LocalStore.getOne(StoreKey.SETTINGS);
+    var shouldOverrideConfig = _.get(settings, mergeKey + ".config", false);
+    var queriesMergeOption = _.get(settings, mergeKey + ".queries", "0");
+
+    var provData = (await LocalStore.getOne(storeKey)) || {};
+
+    if (shouldOverrideConfig) {
+      provData.config = _.get(newData, "Config", {});
+    }
+
+    if (queriesMergeOption === "0") {
```

Comment:
Same as above, `queriesMergeOption ` should contains strings, instead of numbers (so it's more clear to other devs what each value means).